src/main/asciidoc/reference/elasticsearch-repositories.adoc

```diff
@@ -0,0 +1,148 @@
+
+This chapter includes details of the Elasticsearch repository implementation.
+
+include::elasticsearch-repository-queries.adoc[leveloffset=+1]
+
+[[elasticsearch.annotation]]
+== Annotation based configuration
+
+The Spring Data Elasticsearch repositories support can be activated using an annotation through JavaConfig.
+
+.Spring Data Elasticsearch repositories using JavaConfig
+====
+[source,java]
+----
+@Configuration
+@EnableElasticsearchRepositories(                             <1>
+		basePackages = "org/springframework/data/elasticsearch/repositories"
+		)
+static class Config {
+
+    @Bean
+    public ElasticsearchOperations elasticsearchTemplate() {  <2>
+        // ...
+    }
+}
+
+class ProductService {
+
+    private ProductRepository repository;                     <3>
+
+    public ProductService(ProductRepository repository) {
+        this.repository = repository;
+    }
+
+    public Page<Product> findAvailableBookByName(String name, Pageable pageable) {
+        return repository.findByAvailableTrueAndNameStartingWith(name, pageable);
+    }
+}
+----
+
+<1> The `EnableElasticsearchRepositories` annotation activates the Repository support. If no base package is configured, it will use the one of the configuration class it is put on.
+<2> Provide a Bean named `elasticsearchTemplate` of type `ElasticsearchOperations` by using one of the configurations shown in the <<elasticsearch.operations>> chapter.
+<3> Let Spring inject the Repository bean into your class.
+====
+
+[[elasticsearch.cdi]]
+== Elasticsearch Repositories using CDI
+
+The Spring Data Elasticsearch repositories can also be set up using CDI functionality.
+
+.Spring Data Elasticsearch repositories using CDI
+====
+[source,java]
+----
+class ElasticsearchTemplateProducer {
+
+    @Produces
+    @ApplicationScoped
+    public ElasticsearchOperations createElasticsearchTemplate() {
+        // ... <1>
+    }
+}
+
+class ProductService {
+
+    private ProductRepository repository; <2>
+
+    public Page<Product> findAvailableBookByName(String name, Pageable pageable) {
+        return repository.findByAvailableTrueAndNameStartingWith(name, pageable);
+    }
+
+    @Inject
+    public void setRepository(ProductRepository repository) {
+        this.repository = repository;
+    }
+}                
+----
+<1> Create a component by using the same calls as are used in the <<elasticsearch.operations>> chapter.
+<2> Let the CDI framework inject the Repository into your class.
+
+====
+
+[[elasticsearch.namespace]]
+== Spring Namespace
+
+The Spring Data Elasticsearch module contains a custom namespace allowing definition of repository beans as well as elements for instantiating a `ElasticsearchServer` .
+
+Using the `repositories` element looks up Spring Data repositories as described in <<repositories.create-instances>> .
+
+.Setting up Elasticsearch repositories using Namespace
+====
+[source,xml]
+----
+<?xml version="1.0" encoding="UTF-8"?>
```

Comment:
XML examples should make use of indentation.

Reply:
done